### ISSUE TITLE
IDE0130 not showing as warnings in VS 16.10.3

### ISSUE THREAD

**Issue by BahKoo** [dotnet/roslyn#54617]
**Version Used**: 
Visual Studio 2019 16.10.3

**Steps to Reproduce**:
1. In VS 2019 (v 16.10.3), create a new C# project targeting .Net 5.0 
2. Add a new *.editorconfig* file. This automatically includes `dotnet_style_namespace_match_folder = true`
3. Change the namespace of any class so that it doesn't match the folder name

**Expected Behavior**:
The IDE  should show warnings for IDE0130

**Actual Behavior**:
No warnings are shown for IDE0130

I have also tried changing the severity of the rule with each of the following lines in .editorconfig
 - `dotnet_analyzer_diagnostic.severity = warning`
 - `dotnet_diagnostic.IDE0130.severity = warning`
 
I have also tried referencing the `Microsoft.CodeAnalysis.NetAnalyzers` nuget package version 5.0.3.
 
In short, I *think* this is a feature that has been released since VS 2019 16.10.0 but I'm not sure how to get it working. Are there any extra configuration steps that need to be done?

**Comment by Youssef1313** [comment]
@BahKoo Can you try adding `<EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>` under a `PropertyGroup` in your `csproj` file?

In case you see a warning like the following, then this was fixed in 16.11 (https://github.com/dotnet/roslyn/pull/53786):

![image](https://user-images.githubusercontent.com/31348972/124464674-c41c0100-dd94-11eb-8f7d-3eeda87536ae.png)

**Comment by BahKoo** [comment]
@Youssef1313 I'm still not seeing any warnings with `<EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>`. I'm not discounting the possibility that I have another problem with my configuration or with the requirements needed for this warning to occur.

**Comment by BahKoo** [comment]
I have also tested 16.11.0 Preview 2.0 and I can't get an IDE0130 warning in either the IDE or with dotnet build.

**Comment by BahKoo** [comment]
Does anything need to be done to ensure the ProjectDir is available to this analyzer? I see that this is a requirement in the code:
https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/Analyzers/Core/Analyzers/MatchFolderAndNamespace/AbstractMatchFolderAndNamespaceDiagnosticAnalyzer.cs#L52

**Comment by BahKoo** [comment]
I'm now able to get the IDE0130 warnings when I add the following to my .csproj files. Will this be a requirement for the IDE0130 analyzer to work?

```
<ItemGroup>
     <CompilerVisibleProperty Include="ProjectDir" />
</ItemGroup>
```

**Comment by BahKoo** [comment]
It looks like this change was made to the dotnet SDK here but only included in version 6:
https://github.com/dotnet/sdk/commit/e977d0d1f47d18d483513445a5ba37dc0ed23b5d#diff-84d26a3299dcde7ca75281953175d52fb750e945be1ba8ca8f838610ea3a1020

Should this change have been ported to an SDK 5 branch?

**Comment by Youssef1313** [comment]
@ryzngard

**Comment by ryzngard** [comment]
@BahKoo @Youssef1313 these should be in 5.0.300 and higher. @jmarolf ported the PR https://github.com/dotnet/sdk/pull/15836

**Comment by ryzngard** [comment]
@BahKoo do you have a binlog of the build? We can investigate further from there

**Comment by BahKoo** [comment]
@ryzngard I'm not sure what you mean by the binlog. Do you want me to build a solution with the `/bl` flag and give you the output?

Another piece of information that may help is that the following file does not contain the CompilerVisisbleChanges I would expect to see:
`C:\Program Files\dotnet\sdk\5.0.301\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.Analyzers.targets`

**Comment by BahKoo** [comment]
I'm not sure about the branch naming convention but should this have been merged to `release/5.0.3xx` rather than `feed-update-release/5.0.3xx`?

**Comment by ryzngard** [comment]
@BahKoo you can use https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProjectSystemTools2022 to get binlogs (there are 2019 versions as well, depends on what VS you're using) or `/bl` like you mentioned. 

You're right that it does seem to be missing from `5.0.301` on my machine as well. I'm not sure what `feed-update-release` vs `release` is for the sdk :/

**Comment by ryzngard** [comment]
Looks like it was a miss on branch targeting. Jon has taken to fixing for me, thankfully https://github.com/dotnet/sdk/pull/18748

**Comment by Youssef1313** [comment]
@ryzngard I think this can be closed now. The sdk PR got merged.

**Comment by cremor** [comment]
Was there something changed for the rule `dotnet_style_namespace_match_folder`? I can't get it to show up in Visual Studio 2022 v17.0.6. I've tried all the suggestions in the comments here.
